Sources/Basic/TemporaryFile.swift

```diff
@@ -86,31 +108,107 @@ public final class TemporaryFile {
 
     /// Remove the temporary file before deallocating.
     deinit { unlink(path) }
+}
+
+extension TemporaryFile: CustomStringConvertible {
+    public var description: String {
+        return "<TemporaryFile: \(path)>"
+    }
+}
+
+/// FIXME: This isn't right place to declare this, probably POSIX or merge with FSProxyError?
+/// Contains the error which can be thrown while creating a directory using POSIX's mkdir.
```

Comment:
Yeah... I'm not sure how to handle this yet I am still gaining experience with using Swift Errors...

Nit: I don't think FIXME's should be doc comments, usually